Apex/Assignment 5/classes/AttendeeTriggerHandler.cls

```diff
@@ -0,0 +1,60 @@
+public without sharing class AttendeeTriggerHandler {
+    public static void handleAttendeeInsertUpdateAndDelete(List<Attendee__c> newList, Map<Id, Attendee__c> oldMap) {
+        System.debug('***** newList' + newList);
+        System.debug('***** oldMap' + oldMap);
+        Set<Id> contactIds = new Set<Id>();
+        for (Attendee__c att : newList != null ? newList : oldMap.values()) {
+            if (newList != null || oldMap != null) {
+                if (att.Contact__c != null) {
+                    contactIds.add(att.Contact__c);
+                }
+            }
+            if (oldMap != null && oldMap.get(att.Id).Contact__c != null && att.Contact__c != oldMap.get(att.Id).Contact__c) {
+                contactIds.add(oldMap.get(att.Id).Contact__c);
+            }
+        }
+        updateContact(contactIds);
+    }
+    
+    
+    public static void updateContact(Set<Id> contactIds) {
+        System.debug('****** contactIds ' + contactIds);
+        // This Query Provide List of attendees where contact__c value matches contact id in set
```

Comment:
First we need to check that the incoming parameter contactIds is not null or blank and only after that should we do the query.